x/programs/cmd/simulator/src/lib.rs

```diff
@@ -223,109 +225,99 @@ impl Client {
         Self { path }
     }
 
+    fn read_response(
```

Comment:
I don't think this should be a static method, you should just move it to be a normal function as it's private anyway. To be honest, I'm not sure I would put the logic in its own function to begin with. I'll come back here when I see all the places it's being used and leave another comment

Reply:
That's a fair point, my reasoning was that interacting with the simulator should only be done in the `Client`

Reply:
It's used in 3 places already and it's not looking very good when inlined.

Reply:
FYI, I wasn't suggesting you inline the function, I was suggesting to move the function to the module instead of the `mod::read_response` instead of `mod::Client::read_response`.